src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -174,38 +168,49 @@ def q_impl(
         alpha = self.attrs.get("alpha", 1)
         beta = self.attrs.get("beta", 1)
 
+        # If self.constant_inputs is empty this is an encrypted gemm
+        is_encrypted_gemm = isinstance(self.constant_inputs, dict) and not self.constant_inputs
+
         # If alpha != 1 or beta not in [0, 1], this function must be modified
         assert_true(alpha == 1)
-        assert_true(beta in [0, 1])
+        assert_true(beta in {0, 1})
 
         prepared_inputs = self._prepare_inputs_with_constants(
             *q_inputs, calibrate=False, quantize_actual_values=True
         )
 
-        q_input: QuantizedArray = prepared_inputs[0]
-        q_weights: QuantizedArray = prepared_inputs[1]
-        q_bias: Optional[QuantizedArray] = (
-            None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
-        )
+        q_input = prepared_inputs[0]
```

Comment:
maybe `q_input1` ?

Reply:
Done

Reply:
Yes I am going to push sorry. I answer based on my local changed. I then call for a re-review.